core/src/price.ts

```diff
@@ -92,7 +95,7 @@ export const calculateTransactionCollateralOutcome = function (
     return potentialOutcomeCollateralAmount;
 };
 
-export const calculateTransactionGrossProfitDate = function (auction: Auction, currentDate: Date): Date | undefined {
+const _calculateTransactionGrossProfitDate = function (auction: Auction, currentDate: Date): Date | undefined {
```

Comment:
Oh, but you have a `currentDate` as a parameter! That means cache wouldn't make any sense, right?

Reply:
One way around this will be to split this function into two: one that calculates some static numbers based on the `initial price`, `step`, `cut` and `market price` and the other one which applies it. Then you can cache the first one

Reply:
So I am having some issues dealing with this refractor. I currently have the following:

- Determine the initial Price per collaterals (take `initialPrice` and divide it by `collateralAmount`)
- Determine the steps needed to reach profitability based on the marketPrice
- Multiple the amount of steps by the amount of seconds between each step
- Add the number of seconds to the auction start time

Then this value would be cached unless: 
- Step cut/time change (which they should not)
- Market price is adjusted (can happen)

My issue is at the following part. We currently calculate the profitability time with the following function (at least in this PR):
```
const stepNumber = Math.ceil(
  Math.log(auction.marketUnitPrice.dividedBy(auction.approximateUnitPrice).toNumber()) /
  Math.log(auction.priceDropRatio.toNumber())
);
```

In order to cache the value correctly I replaced `auction.approximateUnitPrice` with the initial price per collateral determine before. Afterwards I only needed to change the following part:

```
return addSeconds(currentDate, secondsTillProfitable);
```

Where I changed `currentDate` to `auction.startTime`.

However after doing this I got some very weird numbers from the function. I assume one of the following is happening:
- `initialPrice` is somehow incorrect and therefore the base price per collateral is wrong
- I need to refractor the step count logic (I wouldn't know what tho)


@valiafetisov, maybe you could be so kind to give me some pointers, into which of these issue, it most likely is. Thanks in advance! 😄

Reply:
Ok, let's skip the cache from this PR and do it as a separate issue! Please open it, linking our discussion

Reply:
I created a new issue under #268